### ISSUE TITLE
[Bug] Duckduckgo not working

### ISSUE THREAD

**Issue by jonas-w** [fogine/rofi-google#10]
## With environment variable
Okay i followed your usage guide and set ROFI_SEARCH to ddgr.
When i start rofi-search it displays:
` {"prompt":"search","input action":"send"} `
and waits for input but doesn't open the search gui.
When i type something into the commandline and hit enter this error pops up (e.g. "a"):
``` a
undefined:1
a
^

SyntaxError: Unexpected token a in JSON at position 0
    at JSON.parse (<anonymous>)
    at ReadStream.<anonymous> (/usr/lib/node_modules/rofi-search/rofi-search:80:36)
    at Array.forEach (<anonymous>)
    at ReadStream.<anonymous> (/usr/lib/node_modules/rofi-search/rofi-search:78:22)
    at ReadStream.emit (node:events:527:28)
    at emitReadable_ (node:internal/streams/readable:601:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)

Node.js v17.7.2
```
## Without environment variable
Okay after that i tried it without the environment variable set but no googler installed.
Then the GUI opens and when i type something in it, it immediately crashes and shows this error:
```
Can not find "googler" executable.
BlocksMode-Message: 18:57:03.494: Child 257334 exited abnormally
```
Which makes sense because googler isn't installed. But shouldn't there be a fallback if googler is not installed to use ddgr?

## Information about system and installation
I'm using endeavouros with i3-gaps. I use the linux-zen kernel.
I installed it using AUR but the rofi-blocks-git PKGBUILD file was faulty. So i installed rofi-blocks via `makepkg -si` and an edited version of the PKGBUILD file. More information on that in the AUR Package comments for rofi-blocks-git.

**Comment by fogine** [comment]
I updated `rofi-blocks` aur package.
I see that `googler` github project is archived and its latest version is broken... so I'll need to remove the support for it. For google search you will need to setup google API keys..

How do you execute `rofi-search` exatly?

First simplest method is to
- Execute it just by command `rofi-serach` it will search for `googler` executable and if not found it will try to find `ddgr` executable in your path.. if both executables are not found it will write message to stderr like 
> To make rofi-search work with default configuration, please install either ddgr - https://github.com/jarun/ddgr or googler - https://github.com/jarun/googler

- The second method is explained in the readme in Usage `Examples` section and involves setting env variables and executing rofi search by providing `rofi-search` executable directly to the `rofi`..

**Comment by jonas-w** [comment]
Okay thank you, installing rofi-blocks works now fine. 
I started rofi-search in the command line without the environment variable:
```
Can not find "googler" executable.
BlocksMode-Message: 11:50:15.438: Child 160750 exited abnormally
```
Googler is not installed and ddgr is installed. It seems like it doesn't search for ddgr

**Comment by fogine** [comment]
Hm, thats weird, something is off here, it definitely searchers for `ddgr` [HERE](https://github.com/fogine/rofi-search/blob/master/rofi-search#L736-L745)
I just checked, installed ddgr from aur and it works..
Do you have latest rofi versin? Mine is v `1.7.3`

**Comment by fogine** [comment]
It seams like you have `ROFI_SEARCH` env variable set to `googler` in which case it would fail as you described...

**Comment by jonas-w** [comment]
Yes I have the latest rofi version mine is also `1.7.3`.
I installed rofi-search via aur and ddgr via aur.

Here a gif to show you what happens:
![render1648293428743](https://user-images.githubusercontent.com/32615971/160237207-bb26e0a2-6ddb-479a-afa8-f4291aacc882.gif)

**Comment by jonas-w** [comment]
Okay if no environment variable is set it automatically sets it to googlr with the default config:
https://github.com/fogine/rofi-search/blob/76c7c6f4019944c85a58ea434cd6535985a272f2/rofi-search#L33

**Comment by fogine** [comment]
If you set `ROFI_SEARCH` variable, you need to execute it like this:

```bash
export DDG_ARGS='["-n", 3]'
export ROFI_SEARCH='ddgr'

rofi -modi blocks -blocks-wrap /absolute/path/to/rofi-search -show blocks \ 
-lines 4 -eh 4 -kb-custom-1 'Control+y' -theme /path/to/your/theme.rasi

```

`rofi-search` command works only with default configuration (which in your case does not work for some unknown reason..).

EDIT: you can get default theme for `rofi-search` here: https://gist.github.com/fogine/b79bfdbf48ec70c698c977c13043b816

**Comment by jonas-w** [comment]
Okay thank you with that command it works, but the theme is different from the googlr theme. Is this true?

**Comment by jonas-w** [comment]
Theme is a bit weird. When starting with googler and `rofi-search` it looks normal.
But if i search something with your command the highlighted text i searched nearly has the same color as the background color and the text seems to be invisible
![image](https://user-images.githubusercontent.com/32615971/160238084-e34e6daa-820b-4a74-b1f4-8f055bfd3c00.png)

**Comment by fogine** [comment]
yeah, I'll need to update the default theme, with recent rofi version its broken a little bit..

**Comment by fogine** [comment]
colortheme fixed in `rofi-search@1.1.0` (also removes googler support)
